### PR TITLE
Change usage of buttons

### LINKED ISSUE #400: UdashButton with UdashDropdown
Releated to #379 

`UdashButton` after refactoring does not work well with `UdashDropdown` - it's broken even in the guide example. 

In order to make dropdown buttons look as expected I have to use something like that:
```scala
buttonFactory = content => UdashButton(tag = ButtonTag.Anchor)(nested => Seq[Modifier](content(nested), href := "", tpe := "", cls := "nav-link dropdown-toggle"))
```

### PR DESCRIPTION
**Pull request by mereszeta**
closes #400

### CODE REVIEW

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 1]
```diff
@@ -1,57 +1,78 @@
...
 import org.scalajs.dom._
 import scalatags.JsDom
-import scalatags.JsDom.TypedTag
 import scalatags.JsDom.all._
 
 import scala.scalajs.js
 
+/**
+ * Options for `UdashButton` component
+ * @param color     A button style, one of the standard bootstrap colors `BootstrapStyles.Color`.
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:20]: reformat

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 2]
```diff
@@ -1,57 +1,78 @@
...
+ * @param href            Href attribute value, used only in case of `ButtonTag.Anchor`
+ * @param customModifiers Sequence of custom modifiers.
+ */
+final case class UdashButtonOptions(
+  color: BootstrapStyles.Color = BootstrapStyles.Color.Primary,
+  size: Opt[BootstrapStyles.Size] = Opt.empty,
+  outline: Boolean = false,
+  block: Boolean = false,
+  tag: ButtonTag = ButtonTag.Button,
+  href: Opt[String] = Opt.Empty,
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:34]: If those 2 are used in conjunction, then `ButtonTag.Anchor` should be a class with a `href` instead.

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 3]
```diff
@@ -1,57 +1,78 @@
...
+          onclick :+= ((me: MouseEvent) => if (!disabled.get) fire(ButtonClickEvent(this, me)))
+        )(content(nestedInterceptor)).render
+      case ButtonTag.Anchor =>
+        a(componentId)(classes: _*)(
+          options.href.map(value => href := value)
+        )(content(nestedInterceptor)).render
+      case ButtonTag.Div =>
+        div(componentId)(classes: _*)(
+          onclick :+= ((me: MouseEvent) => if (!disabled.get) fire(ButtonClickEvent(this, me)))
+        )(content(nestedInterceptor)).render
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:74]: `content(nestedInterceptor)).render` can be easily deduplicated (moved to end)

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 4]
```diff
@@ -1,57 +1,78 @@
...
-      onclick :+= ((me: MouseEvent) => if (!disabled.get) fire(ButtonClickEvent(this, me)))
-    )(content(nestedInterceptor)).render
+    options.tag match {
+      case ButtonTag.Button =>
+        button(componentId, tpe := "button")(classes: _*)(
+          onclick :+= ((me: MouseEvent) => if (!disabled.get) fire(ButtonClickEvent(this, me)))
+        )(content(nestedInterceptor)).render
+      case ButtonTag.Anchor =>
+        a(componentId)(classes: _*)(
+          options.href.map(value => href := value)
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:69]: `.map(href := _)`

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 5]
```diff
@@ -1,57 +1,78 @@
...
+
 
   override val render: dom.html.Element = {
-    tag.value(componentId, tpe := "button")(classes: _*)(
-      //condition to support non-button tags
-      onclick :+= ((me: MouseEvent) => if (!disabled.get) fire(ButtonClickEvent(this, me)))
-    )(content(nestedInterceptor)).render
+    options.tag match {
+      case ButtonTag.Button =>
+        button(componentId, tpe := "button")(classes: _*)(
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:64]: Isn't the kiss at the end redundant?

File: bootstrap4/.js/src/test/scala/io/udash/bootstrap/button/UdashButtonTest.scala [hunk 6]
```diff
@@ -76,81 +76,41 @@ class UdashButtonTest extends UdashCoreFrontendTest {
...
 
       disabled.set(true)
 
-      el.classList.length should be(5)
+      el.classList.length should be(6)
       el.classList should contain(BootstrapStyles.active.className)
       el.classList should contain(BootstrapStyles.disabled.className)
       el.classList should contain(BootstrapStyles.Button.btn.className)
       el.classList should contain(BootstrapStyles.Button.block.className)
-      el.classList should contain(BootstrapStyles.Button.outline(BootstrapStyles.Color.Info).className)
```
Comment (ddworak) [inline bootstrap4/.js/src/test/scala/io/udash/bootstrap/button/UdashButtonTest.scala:147]: The "static" options still need to be tested and e.g. outline never is.

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/ButtonsDemo.scala [hunk 7]
```diff
@@ -18,7 +19,7 @@ object ButtonsDemo extends AutoDemo {
 
     import scala.util.Random
 
-    val smallBtn = Some(Size.Small).toProperty[Option[Size]]
+    val smallBtn = Size.Small.opt
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/ButtonsDemo.scala:22]: inline?

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/DatePickerDemo.scala [hunk 8]
```diff
@@ -50,10 +50,10 @@ object DatePickerDemo extends AutoDemo with CssView {
...
     }
 
-    val showButton = UdashButton()("Show")
-    val hideButton = UdashButton()("Hide")
-    val enableButton = UdashButton()("Enable")
-    val disableButton = UdashButton()("Disable")
+    val showButton = UdashButton(options = UdashButtonOptions(color = BootstrapStyles.Color.Secondary))("Show")
+    val hideButton = UdashButton(options = UdashButtonOptions(color = BootstrapStyles.Color.Secondary))("Hide")
+    val enableButton = UdashButton(options = UdashButtonOptions(color = BootstrapStyles.Color.Secondary))("Enable")
+    val disableButton = UdashButton(options = UdashButtonOptions(color = BootstrapStyles.Color.Secondary))("Disable")
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/DatePickerDemo.scala:56]: This "changing to secondary" made the demo snippet uglier. Maybe we can live with the default (whatever it is)?

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala [hunk 9]
```diff
@@ -1,57 +1,78 @@
...
+ * @param color     A button style, one of the standard bootstrap colors `BootstrapStyles.Color`.
+ * @param size            A button size, one of the standard bootstrap sizes `BootstrapStyles.Size`.
+ * @param outline         If true, selects the outline style for the button. More: <a href="http://getbootstrap.com/docs/4.1/components/buttons/#outline-buttons">Bootstrap Docs</a>.
+ * @param block           If true, rendered button will be a full-width block.
+ * @param tag             HTML tag used in button, one of tags defined in `ButtonTag`
+ * @param href            Href attribute value, used only in case of `ButtonTag.Anchor`
+ * @param customModifiers Sequence of custom modifiers.
+ */
+final case class UdashButtonOptions(
+  color: BootstrapStyles.Color = BootstrapStyles.Color.Primary,
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/button/UdashButton.scala:29]: How do I create a button with no button style? I think it's viable and maybe should be default?

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/NavigationDemo.scala [hunk 10]
```diff
@@ -36,11 +39,24 @@ object NavigationDemo extends AutoDemo {
           case MenuContainer(name, children) =>
             val dropdown = UdashDropdown(children.toSeqProperty)(
               (item, _) => linkFactory(item.get),
-              _ => span(name, " ")
+              _ => span(name, " "),
+              buttonFactory = UdashButton(
```
Comment (ddworak) [inline guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/NavigationDemo.scala:43]: I like the code change, but now all those demos became more complex - is it an issue with the API or the snippets?

**Review by ddworak — CHANGES_REQUESTED** [review]

File: guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/NavigationDemo.scala (hunk 10 above)
Comment (mereszeta) [reply guide/guide/.js/src/main/scala/io/udash/web/guide/views/ext/demo/bootstrap/NavigationDemo.scala:44]: I guess that it was necessary to make demos slightly more complicated

**Review by ddworak — APPROVED** [review]
LGTM

**Pull request merged**